spec/models/cart_product_spec.rb

```diff
@@ -0,0 +1,132 @@
+RSpec.describe CartProduct, type: :model do
+  describe 'database' do
+    it 'has columns' do
+      expect(subject).to have_db_column(:quantity).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:total_amount).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:cart_id).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:product_id).of_type(:integer).with_options(null: false)
+      expect(subject).to have_db_column(:created_at).of_type(:datetime).with_options(null: false)
+      expect(subject).to have_db_column(:updated_at).of_type(:datetime).with_options(null: false)
+    end
+
+    it 'has indexes' do
+      expect(subject).to have_db_index(:cart_id)
+      expect(subject).to have_db_index(:product_id)
+    end
+  end
+
+  describe 'associations' do
+    it 'has relationships' do
+      expect(subject).to belong_to(:cart)
+      expect(subject).to belong_to(:product)
+    end
```

Comment:
prefer one `expect` by `it` definition
```suggestion
    it { is_expected.to belong_to(:cart) }
    it { is_expected.to belong_to(:product) }
```

Reply:
Updated.